File: ruby/user_management_organizing_adapter/lib/user_management_organizing/user_management_friendship_checker.rb
require 'organizing/friendship_checker'
require 'user_management/user_id'

module UserManagementOrganizing
  class UserManagementFriendshipChecker
    include Organizing::FriendshipChecker

    def initialize(friendship_repository)
      @friendship_repository = friendship_repository
    end

    def are_friends?(owner_id, friend_id)
      friendships = @friendship_repository.find_by_owner_id(UserManagement::UserId.new(value: owner_id))
      friendships.any? { |f| f.friend_id == UserManagement::UserId.new(value: friend_id) }
    end
  end
end

File: java/user-management-organizing-adapter/src/main/java/com/hexagonalchess/usermanagementorganizing/UserManagementFriendshipChecker.java
package com.hexagonalchess.usermanagementorganizing;

import com.hexagonalchess.organizing.FriendshipChecker;
import com.hexagonalchess.usermanagement.FriendshipRepository;
import com.hexagonalchess.usermanagement.UserId;

public class UserManagementFriendshipChecker implements FriendshipChecker {

    private final FriendshipRepository friendshipRepository;

    public UserManagementFriendshipChecker(FriendshipRepository friendshipRepository) {
        this.friendshipRepository = friendshipRepository;
    }

    @Override
    public boolean areFriends(String ownerId, String friendId) {
        return friendshipRepository.findByOwnerId(new UserId(ownerId)).stream()
            .anyMatch(f -> f.friendId().equals(new UserId(friendId)));
    }
}
